docs/tutorials/client-side-encryption.txt

```diff
@@ -80,40 +90,73 @@ Automatic encryption is a feature that allows users to configure a
 performing database operations. Once the ``Mongo::Client`` is configured, it
 will automatically encrypt any field that requires encryption before writing
 it to the database, and it will automatically decrypt those fields when reading
-them. Automatic encryption is an enterprise-only feature.
+them.
+
+Client-side encryption implements envelope encryption, which is the practice of
+encrypting data with a data key, which is in turn encrypted using a master key.
+Thus, using client-side encryption with MongoDB involves three main steps:
+
+1. Create a master key
+2. Create a data key (and encrypt it using the master key)
+3. Encrypt data using the data key
+
+The example below demonstrates how to follow these steps with a local master key
+in order to perform automatic encryption.
+
+.. note::
+
+  Automatic encryption is an enterprise only feature that only applies to
+  operations on a collection. Automatic encryption is not supported for operations
+  on a database or view, and operations that are not bypassed will result in
+  error (see `Auto Encryption Whitelist <https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#libmongocrypt-auto-encryption-whitelist>`_
+  ). To bypass automatic encryption for all operations, set ``bypass_auto_encryption``
+  to true in ``auto_encryption_options``.
+
+.. note::
 
-The following example provides a demonstration of auto-encryption using a local
-master key.
+  Automatic encryption requires the authenticated user to have the listCollections privilege action.
 
 .. code-block:: ruby
 
   require 'mongo'
 
-  # Generate a local encryption master key
-  # To reuse this master key, persist it to a file or environment variable
-  # on your machine.
+  #####################################
+  # Step 1: Create a local master key #
+  #####################################
+
+  # A local master key is a 96-byte binary blob.
   local_master_key = SecureRandom.random_bytes(96)
+  # => "\xB2\xBE\x8EN\xD4\x14\xC2\x13\xC3..."
```

Comment:
I want to make it clear that this should just be a random binary blob. What is the best way to specify the expected output of this line?

Reply:
I think this is sufficiently clear as it is.